### DISCUSSION TITLE
How to create a new publisher?

### DISCUSSION THREAD

**Discussion by stefan-korn** [GetDKAN/dkan#4056] [Q&A]
Given the case that you start with DKAN from scratch, how do you create a new publisher? When creating a Dataset the publisher is required, but you cannot create a new one with the autocomplete widget.

Is there a way to create just a publisher on its own to reference it later on in the dataset? The same applies for other items like keywords and themes.

**Comment by jastraat** [comment]
You can append the schema you want to use to the node/add/data path and create other types. E.g.

- /node/add/data?schema=publisher
- /node/add/data?schema=keyword

**Comment by stefan-korn** [comment]
Ah, thanks. Good to know that.

**Comment by jastraat** [comment]
@stefan-korn quick adjustment: while the above looks like it would work, the schema query parameter only works with data-dictionaries unfortunately; the identifiers for the other entities types don't generate correctly through the UI subform.

However, you can create new publishers, etc through the larger dataset add form by starting to type and then manually selecting the option that is generated.

**Reply by guille-amarilla** [reply]
@jastraat I'm currently having the same issue, but I can't solve it, where exactly do I have to start typing and then manually selecting? 'Cause all I have in the option for New Publisher is an empty box. I have attached a print screen.
![Screenshot from 2023-12-20 21-25-01](https://github.com/GetDKAN/dkan/assets/154478770/b1087a1d-9908-476a-a9d9-668a387d1db9)

**Reply by jastraat** [reply]
Interesting. That is not the standard display for that form field. Have you customized the schema files or metastore settings? Also, which browser is this in?

**Reply by guille-amarilla** [reply]
Sorry, I downloaded another project by mystake, now I've got the correct one, I can add a Publisher and also add Tags, but when I try to Save the dataset I've got this error message:

![image](https://github.com/GetDKAN/dkan/assets/154478770/466da4c5-dbbe-4ec2-869b-f17e1af300a2)

**Reply by janette** [reply]
@guille-amarilla this means you have entered a string into a field that requires a uri, check your license, Data Dictionary, and Download URL (Link to Data File) values.

**Reply by guille-amarilla** [reply]
Oh, I didn't fill those fields because it's not required, and I'm uploading a file. How can I get around that?

**Comment by stefan-korn** [comment]
> However, you can create new publishers, etc through the larger dataset add form by starting to type and then manually selecting the option that is generated.

Yes, this works. Didn't anticipate that it works like this, not sure why I missed this.

When creating a data dictionary you need to enter an "identifier" which results in being the UUID for the data dictionary then. This works, but the identifier is not neccesarily a valid UUID imho. Using a data-dictionary.ui.json with widget "dkan_uuid" for identifier might solve this. However data-dictionary.ui.json would first need to be added to getAllIds() in SchemaRetriever.

**Reply by janette** [reply]
Yes, we are working on a new data dictionary form that will generate a proper UUID, the current form was an MVP effort. Thanks @stefan-korn .
